### PR TITLE
netex-idfm specs - initial version

### PR DESCRIPTION
**Pull request by prhod**
Work in progress, yet to be done : 

- [x] adding RoutingConstraintZone on stop_times.txt
- [x] checking there is no fare_zone 
- [x] integration of journeyPatternCode and/or train number

### CODE REVIEW

**Comment by prhod** [comment]
This spec requires an evolution of the NTFS: adding a trip_short_name.

File: src/documentation/netex_idfm_to_ntfs.md [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+## Input data description
+This specifications assumes that all the requeried data (time tables for all the lines, stop points and stop areas, transfers, etc.) are provided in one ZIP archive (aka "FICHIERS OFFRE") described in the specification document "NT60-A150701-v1.11-BO-STIF_-_Specification_Technique_d_Interface_NeTEx_pour_la_publication_20190624.docx".
+
+The ZIP archive contains contains: 
+- a **arrets.xml** file
+- a **lignes.xml** file, containing the description of lines, networks and companies
+- a **correspondances.xml** file
+- a folder for each operator (or company) containing:
+  + a **calendriers.xml** file, containing the calendars (or validity patterns) used by trips in files **offre_**
+  + a **commun.xml** file optionnal, containing the comments referenced by the operator objects (if needed)
```
Comment (woshilapin) [inline src/documentation/netex_idfm_to_ntfs.md]:
```suggestion
  + a **commun.xml** file optional, containing the comments referenced by the operator's objects (if needed)
```
?

File: src/documentation/netex_idfm_to_ntfs.md [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+  + a file starting with **offre_** describing the time tables of a specific line
+
+In this document, versions of objects are not handled. The first encountered object description is considered when creating an object.
+
+Each XML files contains a `PublicationDelivery` node, containing a `dataObjects` node. Descriptions below is considering nodes inside this `dataObjects` node.
+
+## Reading of the "arrets.xml" file
+### stops.txt
+`stops` are provided in the **arrets.xml** file in the node **CompositeFrame/frames/GeneralFrame/** (only one **CompositeFrame** is expected). 
+In this netex feed, a `Quay` in included in a "ZDL" `StopPlace`, this "ZDL" `StopPlace` could be included in a "LDA" `StopPlace`.
```
Comment (woshilapin) [inline src/documentation/netex_idfm_to_ntfs.md]:
```suggestion
In this netex feed, a `Quay` is included in a "ZDL" `StopPlace`, this "ZDL" `StopPlace` could be included in a "LDA" `StopPlace`.
```

File: src/documentation/netex_idfm_to_ntfs.md [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+stop_id | *StopPlace/@id* | This field is prefixed. 
+stop_name | *StopPlace/Name* | 
+location_type | | Fixed value `1` (stop_area)
+stop_lat | Quay/Centroid/Location | see (1) below
+stop_lon | Quay/Centroid/Location | see (1) below
+
+(1) Definition of stop_lat et stop_lon:
+As this Netex feed does not provide coordinates for the stop_areas, the stop_lat et stop_lon fields will be set with the coordinate of the centroid of all included stop_points.
+
+(2) Complementary object_properties
```
Comment (papailio) [inline src/documentation/netex_idfm_to_ntfs.md]: Notes 2 and 3 are not referenced in the table above. Maybe separate as subsections?

File: src/documentation/netex_idfm_to_ntfs.md [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+* a ZDL `stop_area` referencing it's `Quay`
+* a LDA `stop_area` contain it's ZDL referencies as complementary object_codes, each ZDL referencing it's `Quay`
+
+
+**Definition of the Accessibility of a stop_point**
+If the `Quay` node contains a `AccessibilityAssessment/MobilityImpairedAccess` node, an equipement will be generated with :
+- a self generated and prefixed `equipment_id`,
+- a `wheelchair_boarding` property set to: 
+  + Fixed value `1` (accessible) if `MobilityImpairedAccess` has the value `true`,
+  + Fixed value `2` (not accessible) if `MobilityImpairedAccess` has the value `true`,
```
Comment (papailio) [inline src/documentation/netex_idfm_to_ntfs.md]: One of the values should be `false`

File: src/documentation/netex_idfm_to_ntfs.md [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+* If source value is "1", then this property is set to "1" (accessible),
+* else his property is set to "0" (unknown)
+
+
+### stop_times.txt
+`stop_times` of a trip are listed in the `passingTimes/TimetabledPassingTime` nodes of a `ServiceJourney` (cf. `trip` definition).
+
+NTFS field | Netex-IDFM element | Mapping rule/Comment
+--- | --- | ---
+trip_id | *Line/@id* | The id of the line is used to create this object. This field is prefixed.
```
Comment (papailio) [inline src/documentation/netex_idfm_to_ntfs.md]: Shouldn't it be *ServiceJourney/@id* as for the trip_id in trips.txt

File: src/documentation/netex_idfm_to_ntfs.md [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+This `RoutingConstraintZone` contains a list of `ScheduledStopPointRef` (in `RoutingConstraintZone/members/ScheduledStopPointRef/@ref`). If a stop_time corresponds to one of those `ScheduledStopPointRef`, the `local_zone_id` is set to the associated integer.
+
+### calendar.txt and calendar_dates.txt
+Active days of `trips` are decribed in **calendriers.xml** of each folder in the `GeneralFrame` node.
+A ValidityPattern is created for each `members/DayType` of the file.
+
+* `ValidBetween/FromDate` and `ValidBetween/ToDate` provide start and end date of each calendars defined in this file (both dates are included). Those dates are restrinctions to be applied to all calendars of this file.
+* A `members/DayType` nodes (referenced by trips), describing the basically active days of a week
+* `members/DayTypeAssignment` nodes, describing exceptions (added or removed days or periods)
+* `members/OperatingPeriod` ndoes, describing periodes referenced in `DayTypeAssignment`
```
Comment (woshilapin) [inline src/documentation/netex_idfm_to_ntfs.md]:
```suggestion
* `members/OperatingPeriod` nodes, describing periods referenced in `DayTypeAssignment`
```

File: src/documentation/netex_idfm_to_ntfs.md [hunk 7]
```diff
@@ -0,0 +1,292 @@
...
+
+In the NTFS, the `service_id` is set to the `members/DayType/@id` attribute (this field is prefixed).
+Active dates are defined by : 
+- Active days described in `members/DayType/properties/PropertyOfDay/DaysOfWeek`. Expected values MUST be one of `Monday`, `Tuesday`, `Wednesday`, `Thursday`, `Friday`, `Saturday`, `Sunday`.
+- One or several `members/DayTypeAssignment`
+  + if the `DayTypeAssignment` contains an attribute `isAvailable` (that should be set at `False`), an inactive date is declared by `DayTypeAssignment/Date`
+  + else an active period referenced by `DayTypeAssignment/OperatingPeriodRef/@ref`. This `OperatingPeriod` is to be used with `DayType` to list effective active dates
+The result is to be restricted between `ValidBetween/FromDate` and `ValidBetween/ToDate` specified at the top level of the file in `GeneralFrame`.
+
+Be careful : Definition of calendars and exceptions in calendar_dates may not be the same definition as the on in the Netex-IDFM files, but the resulting active dates will be the same.
```
Comment (woshilapin) [inline src/documentation/netex_idfm_to_ntfs.md]:
```suggestion
Be careful : Definition of calendars and exceptions in calendar_dates may not be the same definition as the one in the Netex-IDFM files, but the resulting active dates will be the same.
```

File: src/documentation/netex_idfm_to_ntfs.md [hunk 8]
```diff
@@ -0,0 +1,292 @@
...
+line_code | *Line/ShortName* | 
+line_name | *Line/Name* | 
+
+If the node `Line/PrivateCode` is available, the content of this node is added as an `object_code` for this line with `object_system` set at `PrivateCode`.
+
+## Reading of each folder
+In a **offre_** file, 2 **GeneralFrame** are expected in a **CompositeFrame/frames** node:
+* one with a `TypeOfFrameRef/@ref` containing the string `NETEX_STRUCTURE`
+* one with a `TypeOfFrameRef/@ref` containing the string `NETEX_HORAIRE`
+In the following chapters, the **CompositeFrame/frames/GeneralFrame/members** is ommited.
```
Comment (woshilapin) [inline src/documentation/netex_idfm_to_ntfs.md]:
```suggestion
In the following chapters, the **CompositeFrame/frames/GeneralFrame/members** is omitted.
```

File: src/documentation/netex_idfm_to_ntfs.md [hunk 9]
```diff
@@ -0,0 +1,292 @@
...
+`trips` are described in each **offre_** file in the nodes **ServiceJourney**. 
+
+NTFS field | Netex-IDFM element | Mapping rule/Comment
+--- | --- | ---
+route_id | *ServiceJourney/JourneyPatternRef/@ref* | route_id of the Route containing the JourneyPatternRef as an object_code.
+service_id |  | ??? Using DayTypeRef ??? 
+trip_id | *ServiceJourney/@id* | This field is prefixed. 
+trip_headsign | *ServiceJourney/DestinationDisplayRef* | Content of the `DestinationDisplay/FrontText` node. If not available, the name of the `stop_point` of the last `stop_time` is used.
+trip_short_name | *ServiceJourney/DestinationDisplayRef* | Content of the `DestinationDisplay/PublicCode` node. If not available, this field is empty.
+company_id | *ServiceJourney/OperatorRef* | if *ServiceJourney/OperatorRef* is not defined, use *Line/OperatorRef* in *lines.xml* file (cf. lines.txt). This field is prefixed.
```
Comment (woshilapin) [inline src/documentation/netex_idfm_to_ntfs.md]:
```suggestion
company_id | *ServiceJourney/OperatorRef* | if *ServiceJourney/OperatorRef* is not defined, use *Line/OperatorRef* in [*lines.xml*](#linestxt) file. This field is prefixed.
```

File: src/documentation/netex_idfm_to_ntfs.md [hunk 10]
```diff
@@ -0,0 +1,292 @@
...
+physical_mode_id | *Line/TransportMode* | see physical_modes definition
+trip_property_id |  | see trip_properties.txt
```
Comment (woshilapin) [inline src/documentation/netex_idfm_to_ntfs.md]:
```suggestion
trip_property_id |  | see [trip_properties.txt](#trip_propertiestxt)
```

File: src/documentation/netex_idfm_to_ntfs.md [hunk 11]
```diff
@@ -0,0 +1,292 @@
...
+stop_name | *Quay/Name* | 
+location_type | | Fixed value `0` (stop_point)
+stop_lat | Quay/Centroid/Location | see (1) below
+stop_lon | Quay/Centroid/Location | see (1) below
+parent_station | | stop_id of the corresponding `stop_area`, see (2) below.
+stop_timezone | | fixed value `Europe/Paris`
+equipment_id | | This value will be self generated by the connector if an equipment object is necessary. See (2) below.
+
+(1) Definition of stop_lat et stop_lon:
+The `Quay/Centroid/Location` node contains a position (X, Y) with an EPSG described, for exemple:
```
Comment (pascal-vs) [inline src/documentation/netex_idfm_to_ntfs.md]:
```suggestion
The `Quay/Centroid/Location` node contains a position (X, Y) with an EPSG described, for example: 
```

File: src/documentation/netex_idfm_to_ntfs.md [hunk 12]
```diff
@@ -0,0 +1,292 @@
...
+stop_lat | Quay/Centroid/Location | see (1) below
+stop_lon | Quay/Centroid/Location | see (1) below
+parent_station | | stop_id of the corresponding `stop_area`, see (2) below.
+stop_timezone | | fixed value `Europe/Paris`
+equipment_id | | This value will be self generated by the connector if an equipment object is necessary. See (2) below.
+
+(1) Definition of stop_lat et stop_lon:
+The `Quay/Centroid/Location` node contains a position (X, Y) with an EPSG described, for exemple: 
+`<gml:pos srsName="EPSG:2154">662233.0 6861519.0</gml:pos>`
+This coordinate si to be converted to a WGS84 coordinate (EPSG:4326).
```
Comment (pascal-vs) [inline src/documentation/netex_idfm_to_ntfs.md]:
```suggestion
This coordinate is to be converted to a WGS84 coordinate (EPSG:4326).
```

File: src/documentation/netex_idfm_to_ntfs.md [hunk 13]
```diff
@@ -0,0 +1,292 @@
...
+
+(2) Definition of the parent_station:
+A ZDL `StopPlace` includes the referencies of it's ZDE in `StopPlace/quays/QuayRef` nodes. The parent_station of a `stop_point` can be either:
```
Comment (pascal-vs) [inline src/documentation/netex_idfm_to_ntfs.md]:
```suggestion
A ZDL `StopPlace` includes the references of its ZDE in `StopPlace/quays/QuayRef` nodes. The parent_station of a `stop_point` can be either: 
```

File: src/documentation/netex_idfm_to_ntfs.md [hunk 14]
```diff
@@ -0,0 +1,292 @@
...
+equipment_id | | This value will be self generated by the connector if an equipment object is necessary. See (2) below.
+
+(1) Definition of stop_lat et stop_lon:
+The `Quay/Centroid/Location` node contains a position (X, Y) with an EPSG described, for exemple: 
+`<gml:pos srsName="EPSG:2154">662233.0 6861519.0</gml:pos>`
+This coordinate si to be converted to a WGS84 coordinate (EPSG:4326).
+
+(2) Definition of the parent_station:
+A ZDL `StopPlace` includes the referencies of it's ZDE in `StopPlace/quays/QuayRef` nodes. The parent_station of a `stop_point` can be either: 
+* a ZDL `stop_area` referencing it's `Quay`
```
Comment (pascal-vs) [inline src/documentation/netex_idfm_to_ntfs.md]:
```suggestion
* a ZDL `stop_area` referencing its `Quay`
```

File: src/documentation/netex_idfm_to_ntfs.md [hunk 15]
```diff
@@ -0,0 +1,292 @@
...
+* a LDA `stop_area` contain it's ZDL referencies as complementary object_codes, each ZDL referencing it's `Quay`
```
Comment (pascal-vs) [inline src/documentation/netex_idfm_to_ntfs.md]:
```suggestion
* an LDA `stop_area` contains its ZDL references as complementary object_codes, each ZDL referencing its `Quay`
```

File: src/documentation/netex_idfm_to_ntfs.md [hunk 16]
```diff
@@ -0,0 +1,292 @@
...
+
+### routes.txt
+`routes` are provided in each **offre_** file in the nodes **Route**. 
+
+NTFS field | Netex-IDFM element | Mapping rule/Comment
+--- | --- | ---
+route_id | *Route/@id* | This field is prefixed. 
+route_name | *Route/Name* | 
+direction_type | *Route/DirectionType* | The value of this field is used without transformation.
+destination_id |  | The `DirectionRef` of the Route doesn't link to a stop (neither stop_point nor stop_area), thus it's value is not used.
```
Comment (pascal-vs) [inline src/documentation/netex_idfm_to_ntfs.md]:
```suggestion
destination_id |  | The `DirectionRef` of the Route doesn't link to a stop (neither stop_point nor stop_area), thus its value is not used.
```

File: src/documentation/netex_idfm_to_ntfs.md [hunk 17]
```diff
@@ -0,0 +1,292 @@
...
+
+NTFS field | Netex-IDFM element | Mapping rule/Comment
+--- | --- | ---
+trip_property_id | *Line/@id* | The id of the line is used to create this object. This field is prefixed. 
+wheelchair_accessible | `Line/keyList/keyValue/Value` | See (1) below.
+
+wheelchair_accessible value: 
+* If source value is "0", then this property is set to "2" (not accessible),
+* If source value is "1", then this property is set to "1" (accessible),
+* else his property is set to "0" (unknown)
```
Comment (pascal-vs) [inline src/documentation/netex_idfm_to_ntfs.md]:
```suggestion
* else this property is set to "0" (unknown)
```

File: src/documentation/netex_idfm_to_ntfs.md [hunk 18]
```diff
@@ -0,0 +1,292 @@
...
+(3) Definition of local_zone_id
+The declaration of those zones is made in the nodes `RoutingConstraintZone` of the **offre_** file.
+The `local_zone_id` is specified with an auto-incremented integer. Each `RoutingConstraintZone/@id` is associated with a new integer. 
+This `RoutingConstraintZone` contains a list of `ScheduledStopPointRef` (in `RoutingConstraintZone/members/ScheduledStopPointRef/@ref`). If a stop_time corresponds to one of those `ScheduledStopPointRef`, the `local_zone_id` is set to the associated integer.
+
+### calendar.txt and calendar_dates.txt
+Active days of `trips` are decribed in **calendriers.xml** of each folder in the `GeneralFrame` node.
+A ValidityPattern is created for each `members/DayType` of the file.
+
+* `ValidBetween/FromDate` and `ValidBetween/ToDate` provide start and end date of each calendars defined in this file (both dates are included). Those dates are restrinctions to be applied to all calendars of this file.
```
Comment (pascal-vs) [inline src/documentation/netex_idfm_to_ntfs.md]:
```suggestion
* `ValidBetween/FromDate` and `ValidBetween/ToDate` provide start and end dates of each calendar defined in this file (both dates are included). Those dates are restrictions to be applied to all calendars of this file.
```

File: src/documentation/netex_idfm_to_ntfs.md [hunk 19]
```diff
@@ -0,0 +1,292 @@
...
+* A `members/DayType` nodes (referenced by trips), describing the basically active days of a week
+* `members/DayTypeAssignment` nodes, describing exceptions (added or removed days or periods)
+* `members/OperatingPeriod` ndoes, describing periodes referenced in `DayTypeAssignment`
+
+Here is 3 possible modelization in Netex-IDFM of a calendar running from 2016-07-01 to 2016-07-31 except on sundays and on 2016-07-14:
```
Comment (pascal-vs) [inline src/documentation/netex_idfm_to_ntfs.md]:
```suggestion
Here is 3 possible modelizations in Netex-IDFM of a calendar running from 2016-07-01 to 2016-07-31 except on Sundays and on 2016-07-14:
```

**Comment by prhod** [comment]
ready for last review. There may be modifications/improvements in one week or two according to IDFM comments, but I think we can merge it and apply modifications in an other PR. Ok for all ?

File: src/documentation/netex_idfm_to_ntfs.md [hunk 20]
```diff
@@ -0,0 +1,306 @@
...
+![](./netex_idfm_to_ntfs_calendars.png "Definition of calendars in Netex-IDFM specs")
+
+
+### comments.txt
+`comments` are provided in **commun.xml** file in the nodes **GeneralFrame/members/Notice**.
+
+NTFS field | Netex IDFM element | Mapping rule/Comment
+--- | --- | ---
+comment_id | *Notice/@id* | This field is prefixed.
+comment_name | *Notice/Text* |
```
Comment (kinnou02) [inline src/documentation/netex_idfm_to_ntfs.md]: There is no comment type? :(

Comment (prhod) [reply src/documentation/netex_idfm_to_ntfs.md]: There is a `TypeOfNoticeRef`, but in this profile the possible values are `ServiceJourneyNotice` or `LineNotice` only...

File: src/documentation/netex_idfm_to_ntfs.md [hunk 21]
```diff
@@ -0,0 +1,306 @@
...
+
+## Reading of the "arrets.xml" file into stops.txt file
+`stops` are provided in the **arrets.xml** file in the node **CompositeFrame/frames/GeneralFrame/** (only one **CompositeFrame** is expected). 
+In this netex feed, a `Quay` is included in a "ZDL" `StopPlace`, this "ZDL" `StopPlace` could be included in an "LDA" `StopPlace`.
+This connector assumes that all the "ZDL" `StopPlace` are included in one (and only one) "LDA" `StopPlace`.
+
+For french readers: 
+- "ZDE" stands for "Zone D'Embarquement" (a boarding position)
+- "ZDL" stands for "Zone De Lieux" (a group of boarding position with the same transport mode)
+- "LDA" stands for "Lieu D'Arrêt" (a large group of boarding position with serveral transport modes known with a global name)
```
[the remark this answers is not in the document]
Comment (prhod) [reply src/documentation/netex_idfm_to_ntfs.md]: fixed

File: src/documentation/netex_idfm_to_ntfs.md [hunk 22]
```diff
@@ -0,0 +1,306 @@
...
+`trips` are described in each **offre_** file in the nodes **ServiceJourney**. 
+
+NTFS field | Netex-IDFM element | Mapping rule/Comment
+--- | --- | ---
+route_id | *ServiceJourney/JourneyPatternRef/@ref* | route_id of the Route containing the JourneyPatternRef as an object_code.
+service_id |  | Defined using `ServiceJourney/DayTypeRef`, see (calendar.txt and calendar_dates.txt)[]
+trip_id | *ServiceJourney/@id* | This field is prefixed. 
+trip_headsign | *ServiceJourney/DestinationDisplayRef* | Content of the `DestinationDisplay/FrontText` node. If not available, the name of the `stop_point` of the last `stop_time` is used.
+trip_short_name | *ServiceJourney/DestinationDisplayRef* | Content of the `DestinationDisplay/PublicCode` node. If not available, this field is empty.
+company_id | *ServiceJourney/OperatorRef* | if *ServiceJourney/OperatorRef* is not defined, use *Line/OperatorRef* in [*lines.xml*](#linestxt) file. This field is prefixed.f* in *lines.xml* file (cf. lines
```
[the remark this answers is not in the document]
Comment (prhod) [reply src/documentation/netex_idfm_to_ntfs.md]: seems like a big copy/paste error ! thanks

**Pull request merged**